hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -84,16 +77,18 @@ public void doStateTransition() {
 		/* --- Load the model objects --- */
 		final var sender = accountStore.loadAccount(senderId);
 		final var receiver = accountStore.loadContract(contractId);
+		final var callData = !op.getFunctionParameters().isEmpty()
+				? Bytes.fromHexString(CommonUtils.hex(op.getFunctionParameters().toByteArray())) : Bytes.EMPTY;
 
 		/* --- Do the business logic --- */
-		final var txProcessor = new CallEvmTxProcessor(exchange, worldState.updater(), usagePrices, properties);
-		final var result = txProcessor.execute(
+		final var result = evmTxProcessor.execute(
 				sender,
 				receiver,
 				op.getGas(),
 				op.getAmount(),
-				op.getFunctionParameters(),
-				txnCtx.consensusTime());
+				callData,
+				txnCtx.consensusTime(),
+				false);
```

Comment:
Since we **know** that this is not going to be a static call, wouldn't it be better to set it to `false` in the `CallEvmTxProcessor` instead of propagating it through here?